### ISSUE TITLE
Loading BRATS dataset and # of training epochs

### ISSUE THREAD

**Issue by HadiZayer** [shreyaspadhy/UNet-Zoo#12]
I managed to get access of the BRATS dataset, but the scans are in nii format. I managed to use nibabel to load them into a 3d tensors then stacked the different channels. How did you get them in png format? And how did you convert the 3d scans to 2d slices? did you just iterate over the depth of the scan? If so, how did you deal with slices that contain only background?
Also, how many epochs did you need to get these dice scores?

**Comment by HadiZayer** [comment]
I tried to replicate it but for me, the neural network always tend to produce empty results (label everything as background) after few epochs. How did you overcome that?

**Comment by shreyaspadhy** [comment]
Ok let me try answering these one by one - 
1. You don't need the scans to be in png format, as long as you can load them as numpy/PIL Image formats into Python, it doesn't matter. I used MATLAB and the niftii package to save them as .png.
2. Yeah, I iterate over the 3D image, and save every z-coordinate as a 2D image.
3. I chose to ignore all the images that have only background. The totally blank z-coordinates will be at the very top and very bottom of the scan, so you can put some sort of check when you're converting 3D to 2D to only convert if there is a non-zero component in the scan.
4. I believe it took me 100 epochs to get those results. 
5. I believe I was getting completely empty results when I had completely empty images. The way the DICE score is written, it tends to equate any mask to blank with 1.0 score. That's why I added an epsilon to the denominator. If you remove the completely blank scans, your performance should improve. 

Let me know if you need any other help!
